### PR TITLE
Create cleaning skeleton

### LINKED ISSUE #9: Create version of dataset with and without outlier removal
We should then map the entire pipeline to each of these. This is necessary to comply with 
#8

### LINKED ISSUE #10: Restrict datapoint to those who were included at wave 6
> Finally, as the dataset is longitudinal there will be attrition over time. Given that our primary focus is on the health outcomes collected at the sixth time point, we will restrict the dataset to those who were included in Wave 6. We will include a sensitivity analysis which includes all available data. 

Maintain a version of the dataset that includes all data which can be mapped in.

Related: #9 

I suggest we combine these two sensitivity analyses together.

### LINKED ISSUE #14: Clean Movement behaviours data as necessary.
- [x] MVPA
- [x] SED

### LINKED ISSUE #17: Determine whether medical conditions affect any analyses and require observation removal.
> The child has a reported medical condition which would impact on the reasonable evaluation of their health outcomes. For example, we would exclude children with vision loss, or who have restrictions on their participation in physical activity. 

> Where a child’s condition would only meaningfully impact on a subset of the health outcomes (e.g., diabetes), we will exclude the participant only for those analyses.

- [ ]  Fitness
- [ ] Body comp
- [ ] Cardio-metabolic

### LINKED ISSUE #28: Create data cleaning skeleton

### PR DESCRIPTION
**Pull request by tarensanders**
Closes #28, #14, #10, #9

### CODE REVIEW

**Comment by tarensanders** [comment]
Hold off for a sec and I'll incorporate #17 as well

File: R/clean_data.R [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  tidy_df <-
+    full_df %>%
+    dplyr::mutate(
+      indig = dplyr::case_when(
+        stringr::str_detect(indig, "Yes") ~ "Indigenous",
+        indig == "No" ~ "Non-Indigenous",
+        TRUE ~ NA_character_
+      ) %>% as.factor(),
+      age = age_weeks / 52.1775, # Number of weeks in a year
```
Comment (conig) [inline R/clean_data.R:28]: I don't think this matters, but from the dictionary couldn't you grab this without a transformation?
`ff03m1cp` age whole years
`fcage` age decimal

File: R/clean_data.R [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  # Identify kids with health conditions
+  conditions_ids <-
+    dplyr::filter(
+      tidy_df, wave == 6.5,
+      condition_vision == "Not selected",
+      condition_pa == "Not selected",
+      condition_breath == "Not selected",
+      condition_feetlegs == "Not selected"
+    ) %>%
+    dplyr::pull(id)
```
Comment (conig) [inline R/clean_data.R]: Isn't this selecting people without health conditions?

image

File: R/clean_data.R (hunk 1 above)
Comment (tarensanders) [reply R/clean_data.R:28]: The survey data only has weeks, so was just doing it to be consistent since I'll have to calculate it anyway.

**Comment by conig** [comment]
Just waiting on a response to the latter comment before approving.

File: R/clean_data.R (hunk 2 above)
Comment (tarensanders) [reply R/clean_data.R:47]: 'Selected' is if they filled in the box to say that they have the condition. i.e., 13 people have conditions that mean that they have shortness of breath. We want the 1861 who didn't fill in the box.

Comment (tarensanders) [reply R/clean_data.R]: Oh shit yeah you're right.

Comment (tarensanders) [reply R/clean_data.R]: Fixed

**Comment by conig** [comment]
Thanks, looks good

**Review by conig — APPROVED** [review]

**Pull request merged**